pkg/plugins/policies/meshhttproute/xds/filters.go

```diff
@@ -88,6 +88,44 @@ func requestRedirect(redirect api.RequestRedirect, envoyRoute *envoy_route.Route
 			SchemeRedirect: *redirect.Scheme,
 		}
 	}
+	if redirect.Path != nil {
+		switch redirect.Path.Type {
+		case api.ReplaceFullPathType:
+			envoyRedirect.PathRewriteSpecifier = &envoy_route.RedirectAction_RegexRewrite{
+				RegexRewrite: &envoy_type_matcher.RegexMatchAndSubstitute{
+					Pattern: &envoy_type_matcher.RegexMatcher{
+						EngineType: &envoy_type_matcher.RegexMatcher_GoogleRe2{
+							GoogleRe2: &envoy_type_matcher.RegexMatcher_GoogleRE2{},
+						},
+						Regex: `.*`,
+					},
+					Substitution: *redirect.Path.ReplaceFullPath,
+				},
+			}
+		case api.ReplacePrefixMatchType:
```

Comment:
could probably reduce some duplication here

Reply:
yeah, it's not ideal because envoy types are different, but I think I made it more compact